tests/sync/session.cpp

```diff
@@ -335,19 +335,37 @@ TEST_CASE("sync: error handling", "[sync]") {
         CHECK(idx != std::string::npos);
         if (just_before->tm_year == just_after->tm_year) {
             std::stringstream stream;
+#if __GNUC__ < 5
+            char s[5];
+            strftime(s, 5, "%Y", just_after);
```

Comment:
Rather than duplicating this `#if` in  multiple places, why not define a `put_time` helper function that does the right thing for all platforms we use?